app/uk/gov/hmrc/exports/migrations/changelogs/submission/AddActionFieldsForAmend.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.migrations.changelogs.submission
+
+import com.mongodb.client.MongoDatabase
+import com.mongodb.client.model.Updates.{combine, set}
+import org.mongodb.scala.model.{Filters, UpdateOptions}
+import org.mongodb.scala.model.Filters._
+import play.api.Logging
+import uk.gov.hmrc.exports.migrations.changelogs.{MigrationDefinition, MigrationInformation}
+import uk.gov.hmrc.exports.models.declaration.submissions._
+
+import java.util
+import scala.jdk.CollectionConverters._
+
+class AddActionFieldsForAmend extends MigrationDefinition with Logging {
+
+  override val migrationInformation: MigrationInformation =
+    MigrationInformation(
+      id = s"CEDS-4446 Add new fields to the Action model for amended Declarations.",
+      order = 18,
+      author = "Darryl Murphy",
+      runAlways = true
+    )
+  private val uuid = "uuid"
+  private val latestDecId = "latestDecId"
+  private val latestVersionNo = "latestVersionNo"
+
+  override def migrationFunction(db: MongoDatabase): Unit = {
+
+    logger.info(s"Applying '${migrationInformation.id}' db migration...")
+
+    val submissionCollection = db.getCollection("submissions")
+
+    submissionCollection
+      .find(and(exists(latestDecId), exists(uuid), exists(latestVersionNo)))
+      .asScala
+      .map { document =>
+        val submissionId = document.getString(uuid)
+        val submissionLatestDecId = document.getString(latestDecId)
+        val submissionLatestVersionNo = document.getInteger(latestVersionNo)
+
+        submissionCollection.updateOne(
+          Filters.eq(uuid, submissionId),
+          combine(
+            set("actions.$[elem1].decId", submissionId),
+            set("actions.$[elem2].decId", submissionLatestDecId),
+            set("actions.$[elem2].versionNo", submissionLatestVersionNo)
```

Comment:
:thinking: Asking... Is correct to assume that a `CancellationRequest` is always, in the array of `Action`s, the 2nd element?
Maybe I'm saying something wrong because now I do not remember well, but isn't the 1st element of the `Action` array always the last-in-time action inserted? If thats the case, you could also have...
```
  set("actions.$[elem1].decId", submissionLatestDecId),
  set("actions.$[elem2].decId", submissionId),
```

Reply:
`elem2` is a given name, it could quite easily be `elemCancellation` or whatever

Reply:
I don't quite understand your answer here

Reply:
I'll make a change to demonstrate